core/src/main/java/hudson/model/listeners/ItemListener.java

```diff
@@ -57,6 +58,18 @@ public void onCreated(Item item) {
     }
 
     /**
+     * Called before a job is copied into a new parent, providing the ability to veto the copy operation before it
+     * starts.
+     *
+     * @param src the item being copied
+     * @param parent the proposed parent
+     * @throws Failure to veto the operation.
+     * @since TODO
+     */
+    public void onCheckCopy(Item src, ItemGroup parent) throws Failure {
```

Comment:
I am not a big fan of it since it is a `RuntimeException`, hence we potentially get into the error propagation fun. But I will be fine if you create an equivalent non-Runtime exception with support of the error rendering